visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -100,115 +104,227 @@ export class NodeDecorator {
-						node.data.edgeAttributes[name].outgoing += value.outgoing
+			for (const name of attributeKeys) {
+				const selector = `${name}${data.path}`
+				const parentSelector = `${name}${parent.data.path}`
+				if (attributeTypes.nodes[name] === AttributeTypeValue.relative) {
+					setNodeMediansToParent(medians, selector, parentSelector, data, name, isDeltaState)
+					collectNodeMediansOnParent(medians, parentSelector, data, name, isDeltaState)
+				} else {
+					parent.data.attributes[name] += data.attributes[name]
+					if (isDeltaState) {
+						parent.data.deltas[name] = parent.data.deltas[name] ?? 0
+						parent.data.deltas[name] += data.deltas[name] ?? 0
 					}
 				}
 			}
 
-			for (const { name } of metricData) {
-				data.attributes[name] = this.aggregateLeafMetric(leafAttributes, name, attributeTypes)
-				if (isDeltaState) {
-					data.deltas[name] = this.aggregateLeafMetric(leafDeltas, name, attributeTypes)
+			for (const name of edgeKeys) {
+				const value = data.edgeAttributes[name]
+
+				if (!value) {
+					continue
 				}
-			}
 
-			if (edgeMetrics) {
-				for (const [name, value] of Object.entries(edgeMetrics)) {
-					const temporary = value as Record<string, number[]>
-					data.edgeAttributes[name].incoming = this.median(temporary.incoming)
-					data.edgeAttributes[name].outgoing = this.median(temporary.outgoing)
+				const selector = `${name}${data.path}`
+				const parentSelector = `${name}${parent.data.path}`
+				if (attributeTypes.edges[name] === AttributeTypeValue.relative) {
+					setEdgeMediansToParent(
+						medians,
+						`${MedianSelectors.INCOMING}${selector}`,
+						`${MedianSelectors.INCOMING}${parentSelector}`,
+						data,
+						name,
+						EdgeAttributeType.INCOMING
+					)
+					setEdgeMediansToParent(
+						medians,
+						`${MedianSelectors.OUTGOING}${selector}`,
+						`${MedianSelectors.OUTGOING}${parentSelector}`,
+						data,
+						name,
+						EdgeAttributeType.OUTGOING
+					)
+					collectEdgeMediansOnParent(
+						medians,
+						`${MedianSelectors.INCOMING}${parentSelector}`,
+						data,
+						name,
+						EdgeAttributeType.INCOMING
+					)
+					collectEdgeMediansOnParent(
+						medians,
+						`${MedianSelectors.OUTGOING}${parentSelector}`,
+						data,
+						name,
+						EdgeAttributeType.OUTGOING
+					)
+				} else {
+					parent.data.edgeAttributes[name].incoming += value.incoming
+					parent.data.edgeAttributes[name].outgoing += value.outgoing
 				}
 			}
 		})
-		return map
-	}
 
-	private static aggregateLeafMetric(metrics: KeyValuePair[], metricName: string, attributeTypes: AttributeTypes) {
-		if (attributeTypes.nodes[metricName] === AttributeTypeValue.relative) {
-			const metricValues: number[] = []
-			for (const metric of metrics) {
-				const number = metric[metricName]
-				if (number) {
-					metricValues.push(number)
+		// Set Median for root node
+		for (const name of edgeKeys) {
+			if (attributeTypes.edges[name] === AttributeTypeValue.relative) {
```

Comment:
I suggest we optimize this a tiny bit by collecting the relative edgeMetricKeys up front and the same with the attributeKeys that are relative. That way there's no need to check for those on each node as we already know the correct names.